lib/graphql/schema/build_from_definition.rb

```diff
@@ -247,13 +258,16 @@ def build_default_value(default_value)
           end
         end
 
+        NO_DEFAULT_VALUE = {}.freeze
+
         def build_arguments(type_class, arguments, type_resolver)
           builder = self
 
           arguments.each do |argument_defn|
-            default_value_kwargs = {}
-            if !argument_defn.default_value.nil?
-              default_value_kwargs[:default_value] = builder.build_default_value(argument_defn.default_value)
+            default_value_kwargs = if !argument_defn.default_value.nil?
+              { default_value: builder.build_default_value(argument_defn.default_value) }
+            else
+              NO_DEFAULT_VALUE
```

Comment:
It turned out this was getting retained by the `define_method` call in `input_object.rb`, I could probably revert these changes, but it seems good anyways to reduce the allocated Hashes.